src/IonEvent.ts

```diff
@@ -482,20 +538,20 @@ class IonSexpEvent extends AbsIonContainerEvent {
 
     }
 
-    valueEquals(expected: IonSexpEvent): boolean {
-        if (!(expected instanceof IonSexpEvent)) return false;
+    valueCompare(expected: IonSexpEvent): ComparisonResult {
+        if (!(expected instanceof IonSexpEvent)) return new ComparisonResult(ComparisonResultType.NOT_EQUAL,"Event types don't match");
         let container = this.ionValue;
         let expectedContainer = expected.ionValue;
-        if (container.length !== expectedContainer.length) return false;
+        if (container.length !== expectedContainer.length) return new ComparisonResult(ComparisonResultType.NOT_EQUAL, "S-expression length don't match");
         for (let i: number = 0; i < container.length; i++) {
             let child = container[i];
-            if (!child.equals(expectedContainer[i])) {
-                return false;
+            if (child.compare(expectedContainer[i]).result == ComparisonResultType.NOT_EQUAL) {
+                return child.compare(expectedContainer[i]);
```

Comment:
Looks like this compare is performed twice. Consider storing the result and returning it later.